src/parser/regexp.bnf

```diff
@@ -694,6 +703,8 @@ NonemptyClassRanges
 
   | ClassAtom DASH ClassAtom ClassRanges
     {
+      checkClassRange($1.value, $3.value);
```

Comment:
👍 We should also call this check for the `NonemptyClassRangesNoDash : ClassAtomNoDash DASH ClassAtom ClassRanges` production below.

You can test this by two examples:

```
[a-z]
[-a-z]
```